src/Compilers/CSharp/Portable/Symbols/Source/FieldSymbolWithAttributesAndModifiers.cs

```diff
@@ -276,22 +280,26 @@ private void VerifyConstantValueMatches(ConstantValue attrValue, ref DecodeWellK
             if (!attrValue.IsBad)
             {
                 var data = arguments.GetOrCreateData<FieldWellKnownAttributeData>();
-                ConstantValue constValue;
+                ConstantValue? constValue;
 
                 if (this.IsConst)
                 {
                     if (this.Type.SpecialType == SpecialType.System_Decimal)
                     {
                         constValue = this.GetConstantValue(ConstantFieldsInProgress.Empty, earlyDecodingWellKnownAttributes: false);
 
-                        if ((object)constValue != null && !constValue.IsBad && constValue != attrValue)
+                        if ((object?)constValue != null && !constValue.IsBad && constValue != attrValue)
                         {
+#nullable disable // Can 'arguments.AttributeSyntaxOpt' be null? https://github.com/dotnet/roslyn/issues/39166
```

Comment:
I don't think it can be null. Looking at its documentation, it says that it won't be null when dealing with attributes in source, which is what this method deals with.
I'd remove the `#nullable disable` and use an assertion instead.